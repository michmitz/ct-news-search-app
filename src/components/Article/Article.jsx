import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, author, description }) => (
  <>
    <h1>{title}</h1>
    <h3>{author}</h3>
    <p>{description}</p>
  </>
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default Article;