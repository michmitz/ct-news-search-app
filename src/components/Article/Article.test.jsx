import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Article from './Article'

describe('Article component', () => {
  afterEach(() => cleanup());
  it('renders article', () => {
    const { asFragment } = render(<Article
      title="Test Title"
      author="test-author"
      description="test-description"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
