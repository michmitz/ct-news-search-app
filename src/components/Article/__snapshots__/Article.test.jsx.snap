// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders article 1`] = `
<DocumentFragment>
  <h1>
    Test Title
  </h1>
  <h3>
    test-author
  </h3>
  <p>
    test-description
  </p>
</DocumentFragment>
`;
